tripal_chado/src/Plugin/TripalBackendPublish/ChadoPublish.php

```diff
@@ -1316,6 +1398,35 @@ protected function publish_summarize(bool $success, array $stats): void {
     $this->logger->notice($message);
   }
 
+  /**
+   * Generates the options array for chado storage findValues().
+   *
+   * These options include appropriate max_delta specifications.
+   * The cardinality value, if a positive integer greater than 1,
+   * will override the global setting.
+   *
+   * @return array
+   *   The options to pass to findValues().
+   */
+  protected function getFindValuesOptions(): array {
+    // We add one to max_delta to use the extra record as a flag that
+    // at least one record was not published. The formatters will not
+    // display this last extra record.
+    $max_deltas = [];
+    foreach ($this->field_info as $field_name => $field_info) {
+      if (array_key_exists('object_table', $field_info)) {
+        if ($field_info['cardinality'] > 1) {
+          $max_deltas[$field_info['object_table']] = $field_info['cardinality'] + 1;
```

Comment:
This is going to cause inconsistent behaviour where there are multiple fields with the same object table and different cardinality :see_no_evil: For example, with property fields, if field A has a cardinality of 5 and field B has a cardinality of 10 then depending on order we might get a max delta of 6 or 11... I wonder if we can overcome this for such fields by finding the table alias here since property fields will alias the tables to keep them separate.

Reply:
Hmm, I kind of forgot about properties, and max delta is not implemented for those because they are "single hop". There is more to do...

Reply:
Another example of this would be the contact by role field. This will alias the linking table so as to keep the fields apart based on the type of link.

Reply:
Based on code-review this looks resolved as well.